common/plugins/eu.esdihumboldt.hale.common.instance/src/eu/esdihumboldt/hale/common/instance/processing/InstanceProcessor.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2017 wetransform GmbH
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
+
+package eu.esdihumboldt.hale.common.instance.processing;
+
+import eu.esdihumboldt.hale.common.core.service.ServiceProvider;
+import eu.esdihumboldt.hale.common.instance.model.Instance;
+import eu.esdihumboldt.hale.common.instance.model.InstanceReference;
+
+/**
+ * {@link Instance} processor
+ * 
+ * @author Florian Esser
+ */
+public interface InstanceProcessor {
```

Comment:
Some instance processors might need the information when the processing (of the current batch of instances) is finished. The processor might for instance want to free up resources that it allocated. Do you think that could be easily added? We could use the `Closable` interface to extend from.